slack/lib.w

```diff
@@ -0,0 +1,72 @@
+bring cloud;
+bring http;
+bring "./util.w" as helpers;
+bring "./message.w" as msg;
+bring "./events.w" as events;
+bring "./context.w" as context;
+bring "cdktf" as cdktf;
+bring util;
+bring ui;
+
+/// Properties for Slack bot
+pub struct SlackbotProps {
+  ignoreBots: bool?;
+}
+
+
+pub class Slackbot {
+  api: cloud.Api;
+  onEventHandlers: MutMap<inflight(context.CtxEventCallBack, Json):void>;
+  ignoreBots: bool;
+  botToken: cloud.Secret;
+
+  new(props: SlackbotProps) {
+    this.onEventHandlers = MutMap<inflight (context.CtxEventCallBack, Json): void>{};
+    this.ignoreBots = props?.ignoreBots ?? true;
+    this.botToken = new cloud.Secret(name: "{this.node.id}_botToken");
+    this.api = new cloud.Api();
+
+    if util.env("WING_TARGET") == "tf-aws" {
```

Comment:
Technically this is harmless in the simulator so the if statement is not really needed.